src/KafkaW/Consumer.cpp

```diff
@@ -127,17 +133,36 @@ bool Consumer::topicPresent(const std::string &TopicName) {
   return true;
 }
 
+/// Updates the RdKafka Metadata pointer. If broker is unavailable, keeps trying
+/// to connect every 500ms, logging messages every few seconds.
 void Consumer::updateMetadata() {
+  short LoopCounter = 0;
+  while (!metadataCall()) {
+    if (LoopCounter == 7) {
+      Logger->error("Cannot contact broker, retrying until connection is "
+                    "established...");
+      LoopCounter = 0;
+    }
+    LoopCounter++;
+  }
+  Logger->info("Connection with broker established.");
+}
+
+bool Consumer::metadataCall() {
   RdKafka::Metadata *MetadataPtr = nullptr;
   auto ErrorCode = KafkaConsumer->metadata(
       true, nullptr, &MetadataPtr, ConsumerBrokerSettings.MetadataTimeoutMS);
-  if (ErrorCode == RdKafka::ERR__TRANSPORT)
-    throw MetadataException("Cannot contact broker");
-  else if (ErrorCode != RdKafka::ERR_NO_ERROR) {
+  switch (ErrorCode) {
+  case RdKafka::ERR_NO_ERROR:
+    KafkaMetadata = std::shared_ptr<RdKafka::Metadata>(MetadataPtr);
+    return true;
+  case RdKafka::ERR__TRANSPORT:
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
Can you tell me the reasoning for sleeping here?

Reply:
I wanted to pause the thread only if this particular error occurs. If everything goes right or if another error happens there is no delay.

Reply:
@mchorazak Can you be a bit more specific as to why? :-)

Reply:
The idea was to avoid spamming metadata requests if it fails, but it looks like `RdKafka::ERR__TRANSPORT` only occurs if the timeout is hit anyway.